python/tests/test_telemetry.py

```diff
@@ -0,0 +1,47 @@
+import pytest
+import requests
+
+import lancedb
+from lancedb.utils.events import _Events
+from lancedb.utils.general import threaded_request
+
+
+def mock_request(
+    method, url, retry=3, timeout=30, thread=True, code=-1, verbose=True, **kwargs
+):
+    assert method.lower() == "post"
+    json = kwargs.get("json", {})
+    # TODO: don't hardcode these here. Instead create a module level json scehma in lancedb.utils.events for better evolvability
+    batch_keys = ["api_key", "distinct_id", "batch"]
+    event_keys = ["event", "properties", "timestamp", "distinct_id"]
+    property_keys = ["cli", "install", "platforms", "version", "session_id", "blud"]
```

Comment:
Added loose tests for now here. Ideally we could hardcode the json schema as global var and directly assert here but I don't want to edit the reviewed files so will do in another PR